src/main/java/org/radarcns/webapp/resource/AppStatusEndPoint.java

```diff
@@ -80,20 +81,16 @@
     @ApiResponse(responseCode = "403", description = "Not Authorised error occurred")
     @ApiResponse(responseCode = "404", description = "Subject not found.")
     @NeedsPermissionOnSubject(entity = SOURCE, operation = READ)
-    public Application getLastReceivedAppStatusJson(
+    public MonitorData getLastReceivedAppStatusJson(
             @Alphanumeric @PathParam(PROJECT_NAME) String projectName,
             @Alphanumeric @PathParam(SUBJECT_ID) String subjectId,
             @Alphanumeric @PathParam(SOURCE_ID) String sourceId) throws IOException {
         mpClient.checkSubjectInProject(projectName, subjectId);
-        Application application = null;
+        MonitorData result = null;
         if (subjectService.checkSourceAssignedToSubject(subjectId, sourceId)) {
-            application = applicationStatusMonitorService.getStatus(projectName,
-                    subjectId, sourceId, mongoClient);
-        }
-        if (application == null) {
-            return new Application(null, 0d, ServerStatus.UNKNOWN, -1, -1, -1);
-
+            result = sourceStatusMonitorService.getStatus(projectName, subjectId, sourceId,
+                    mongoClient);
         }
-        return application;
+        return result;
```

Comment:
this will return null. Perhaps you could return 404 for missing subjects? The subjectService has a weird syntax though... It returns `boolean` but never returns `false`, instead throws a `BadRequestException` where I'd expect a `404`. Perhaps it can return `void` and map to `NotFoundException`?

Reply:
The `checkSubjectInProject` should return 404 when the subject not found as well as when not found the project.
I will change the subjectService